cli/src/execDocker/docker.js

```diff
@@ -31,71 +32,88 @@ export async function dockerBuild({
   if (osType === 'Darwin' && isForTest) {
     platform = 'linux/arm64';
   }
-
+  const abortSignal = createAbortSignal();
   // Perform the Docker build operation
   const buildImageStream = await docker.buildImage(buildArgs, {
     t: tag,
     platform,
     pull: true, // docker store does not support multi platform image, this can cause issues when switching build target platform, pulling ensures the right image is used
+    abortSignal,
   });
 
-  const imageId = await new Promise((resolve, reject) => {
-    docker.modem.followProgress(buildImageStream, onFinished, onProgress);
-
-    function onFinished(err, output) {
-      /**
-       * expected output format for image id
-       * ```
-       *   {
-       *    aux: {
-       *      ID: 'sha256:e994101ce877e9b42f31f1508e11bbeb8fa5096a1fb2d0c650a6a26797b1906b'
-       *    }
-       *  },
-       * ```
-       */
-      const builtImageId = output?.find((row) => row?.aux?.ID)?.aux?.ID;
+  let imageId = null;
 
-      /**
-       * 3 kind of error possible, we want to catch each of them:
-       * - stream error
-       * - build error
-       * - no image id (should not happen)
-       *
-       * expected output format for build error
-       * ```
-       *   {
-       *     errorDetail: {
-       *       code: 1,
-       *       message: "The command '/bin/sh -c npm ci' returned a non-zero code: 1"
-       *     },
-       *     error: "The command '/bin/sh -c npm ci' returned a non-zero code: 1"
-       *   }
-       * ```
-       */
-      const errorOrErrorMessage =
-        err || // stream error
-        output.find((row) => row?.error)?.error || // build error message
-        (!builtImageId && 'Failed to retrieve generated image ID'); // no image id -> error message
+  try {
+    const imageId = await new Promise((resolve, reject) => {
+      // Handle abort signal
+      if (abortSignal) {
+        abortSignal.addEventListener('abort', () => {
+          buildImageStream.destroy();
+          reject(new Error('Docker build aborted'));
+        });
+      }
```

Comment:
this seems to be useless, if `abortSignal` is triggered, `buildImage` will stop the build with an error `Error: aborted` no need to destroy the stream or to reject the promise, the default handler will do it.